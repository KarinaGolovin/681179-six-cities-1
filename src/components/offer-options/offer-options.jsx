import React from 'react';
import PropTypes from 'prop-types';
import {capitalizeFirstLetter} from '../../utils';

export const OfferOptions = ({offerType, offerBedrooms, offerCapacity}) => {
  const OPTIONS = [
    {
      optionClass: `property__feature--entire`,
      optionText: capitalizeFirstLetter(offerType),
    },
    {
      optionClass: `property__feature--bedrooms`,
      optionText: `${offerBedrooms} Bedrooms`
    },
    {
      optionClass: `property__feature--adults`,
      optionText: `Max ${offerCapacity} adults`
    },
  ];

  return (
    <ul className="property__features">
      {OPTIONS.map((option) => {
        return (
          <li className={`property__feature ${option.optionClass}`} key={option.optionClass}>
            {option.optionText}
          </li>
        );
      })}
    </ul>
  );
};

OfferOptions.propTypes = {
  offerType: PropTypes.string,
  offerBedrooms: PropTypes.number,
  offerCapacity: PropTypes.number,
};
