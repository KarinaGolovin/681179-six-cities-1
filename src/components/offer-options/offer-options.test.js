import React from 'react';
import renderer from 'react-test-renderer';
import {OfferOptions} from './offer-options.jsx';

it(`OfferOptions correctly renders`, () => {
  const component = renderer.create(
      <OfferOptions
        offerType={`test`}
        offerBedrooms={1}
        offerCapacity={1}
      />
  ).toJSON();

  expect(component).toMatchSnapshot();
});
