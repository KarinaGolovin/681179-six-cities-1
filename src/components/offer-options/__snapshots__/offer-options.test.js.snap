// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferOptions correctly renders 1`] = `
<ul
  className="property__features"
>
  <li
    className="property__feature property__feature--entire"
  >
    Test
  </li>
  <li
    className="property__feature property__feature--bedrooms"
  >
    1 Bedrooms
  </li>
  <li
    className="property__feature property__feature--adults"
  >
    Max 1 adults
  </li>
</ul>
`;
